classes/session/SessionManager.inc.php

```diff
@@ -38,12 +38,18 @@ function __construct($sessionDao, $request) {
 		ini_set('session.name', Config::getVar('general', 'session_cookie_name')); // Cookie name
 		ini_set('session.cookie_lifetime', 0);
 		ini_set('session.cookie_path', Config::getVar('general', 'session_cookie_path', $request->getBasePath() . '/'));
-		ini_set('session.cookie_domain', $request->getServerHost(null, false));
 		ini_set('session.gc_probability', 1);
 		ini_set('session.gc_maxlifetime', 60 * 60);
 		ini_set('session.auto_start', 1);
 		ini_set('session.cache_limiter', 'none');
 
+		// Allow cookie_domain override from config.inc.php
+		if (Config::getVar('general', 'session_cookie_domain')){
```

Comment:
You can provide a default in the optional 3rd parameter of `Config::getVar`, i.e. I think you could replace this entire block with:
```
ini_set(
  'session.cookie_domain',
  Config::getVar('general', 'session_cookie_domain', $request->getServerHost(null, false))
);
```

Reply:
I have a potential solution below that would not require any new settings and should work with any combination of domains.